### PR TITLE
#1275 fix 'couldn't find active FES url' startup error

### LINKED ISSUE #1275: 'Couldn't get active FES url' error on startup
@tomholub I tried to login to outlook account and get the error
Screenshot 2022-01-05 at 16 26 49
can you check pls?

_Originally posted by @fcvakintos in https://github.com/FlowCrypt/flowcrypt-ios/issues/484#issuecomment-1005732008_

### PR DESCRIPTION
**Pull request by sosnovsky**
This PR fixes 'Couldn't find active FES url' error on startup

close #1275
----------------------------------

**Tests** _(delete all except exactly one)_:
- Tests added or updated - added `testGetClientConfigurationWithoutActiveFesUrl`

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [x] addresses the issue it closes (if any)
- [x] code is readable and understandable
- [x] is accompanied with tests, or tests are not needed
- [x] is free of vulnerabilities
- [x] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

**Review by tomholub — APPROVED** [review]
:+1:

File: FlowCrypt/Functionality/Services/Account Server Services/EnterpriseServerApi.swift [hunk 1]
```diff
@@ -101,14 +101,21 @@ class EnterpriseServerApi: EnterpriseServerApiType {
...
+        do {
+            let response: ClientConfigurationResponse = try await performRequest(
+                email: email,
+                url: "/api/v1/client-configuration?domain=\(userDomain)",
+                method: .get,
+                withAuthorization: false
+            )
+
+            return response.clientConfiguration
+        } catch EnterpriseServerApiError.noActiveFesUrl {
```
Comment (Kharchevskyi) [inline FlowCrypt/Functionality/Services/Account Server Services/EnterpriseServerApi.swift:113]: Maybe it's better to not throw error in `getActiveFesUrl` for `publicEmailProviderDomains`

**Comment by Kharchevskyi** [comment]
Also it looks that it's impossible to log out if any error occurs during app startup. Please check 
#1267

**Comment by tomholub** [comment]
> Also it looks that it's impossible to log out if any error occurs during app startup. Please check #1267 There is no other way than delete app if it happens

I'll track that as a separate issue

File: FlowCrypt/Functionality/Services/Account Server Services/EnterpriseServerApi.swift (hunk 1 above)
Comment (sosnovsky) [reply FlowCrypt/Functionality/Services/Account Server Services/EnterpriseServerApi.swift:113]: I think we can leave `throw` there, but improve it's handling, so users will be able to re-login after startup error

Comment (tomholub) [reply FlowCrypt/Functionality/Services/Account Server Services/EnterpriseServerApi.swift:113]: Let's leave the throw for now and improve handling as Roman says. We could re-visit it in the future.

**Pull request merged**